app/src/main/java/ru/alexzdns/fundamentals/homework/ui/movieDetails/MovieDetailsViewModelFactory.kt

```diff
@@ -0,0 +1,14 @@
+package ru.alexzdns.fundamentals.homework.ui.movieDetails
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import ru.alexzdns.fundamentals.homework.network.MovieApi
+import ru.alexzdns.fundamentals.homework.network.NetworkModule
+
+class MovieDetailsViewModelFactory(): ViewModelProvider.Factory {
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T = when (modelClass) {
+        MovieDetailsViewModel::class.java -> MovieDetailsViewModel(NetworkModule.retrofit.create(MovieApi::class.java))
```

Comment:
Есть еще один вариант метода create(). Компилятор Котлин умеет приводить к нужному типу.
```suggestion
        MovieDetailsViewModel::class.java -> MovieDetailsViewModel(NetworkModule.retrofit.create())
```